src/main/java/me/anxuiz/settings/bukkit/plugin/gui/clickables/Clickable.java

```diff
@@ -0,0 +1,28 @@
+package me.anxuiz.settings.bukkit.plugin.gui.clickables;
+
+import me.anxuiz.settings.bukkit.plugin.BukkitSettingsPlugin;
+import org.bukkit.Bukkit;
+import org.bukkit.event.Listener;
+import org.bukkit.inventory.ItemStack;
+
+public abstract class Clickable implements Click, Listener {
```

Comment:
There are a lot of things wrong with this class.

Firstly, it's absurdly inefficient. Every click event on the server is dispatched to every item in every open GUI. Even events from other types of inventories.

Why does the base class register for events, but it relies on the subclass to unregister? Individual click handlers shouldn't have to deal with that.

And the way your click handlers are unregistering is completely broken. When any type of inventory closes, by any player anywhere on the server, all GUIs that happen to be open at the time will unregister themselves and become broken.

And even if they checked that the right inventory is being closed, it would still be really easy to leak listeners. If you create one of these things, and it doesn't end up in an open GUI for whatever reason, it stays registered forever.